app/src/main/java/com/google/samples/apps/sunflower/compose/plantdetail/PlantDetailView.kt

```diff
@@ -0,0 +1,425 @@
+import androidx.ui.material.IconButton
+import androidx.ui.material.MaterialTheme
+import androidx.ui.material.ProvideEmphasis
+import androidx.ui.material.Surface
+import androidx.ui.material.TopAppBar
+import androidx.ui.material.icons.Icons
+import androidx.ui.material.icons.filled.Add
+import androidx.ui.material.icons.filled.ArrowBack
+import androidx.ui.material.icons.filled.Share
+import androidx.ui.res.stringResource
+import androidx.ui.semantics.accessibilityLabel
+import androidx.ui.text.font.FontWeight
+import androidx.ui.tooling.preview.Preview
+import androidx.ui.unit.Dp
+import androidx.ui.unit.dp
+import androidx.ui.viewinterop.AndroidView
+import androidx.ui.viewmodel.viewModel
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.compose.TextSnackbarContainer
+import com.google.samples.apps.sunflower.compose.getQuantityString
+import com.google.samples.apps.sunflower.compose.systemBarsPadding
+import com.google.samples.apps.sunflower.compose.visible
+import com.google.samples.apps.sunflower.data.Plant
+import com.google.samples.apps.sunflower.utilities.InjectorUtils
+import com.google.samples.apps.sunflower.viewmodels.PlantDetailViewModel
+import dev.chrisbanes.accompanist.coil.CoilImageWithCrossfade
+import dev.chrisbanes.accompanist.mdctheme.MaterialThemeFromMdcTheme
+
+
+/**
+ * As these callbacks are passed in through multiple Composables, to avoid having to name
+ * parameters to not mix them up, they're aggregated in this class.
+ */
+data class PlantDetailsCallbacks(
+    val onFabClicked: () -> Unit,
+    val onBackClicked: () -> Unit,
+    val onShareClicked: () -> Unit
+)
+
+@Composable
+fun PlantDetailsScreen(
+    plantId: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: (String) -> Unit
+) {
+    // ViewModel and LiveDatas needed to populate the plant details info on the screen
+    val plantDetailsViewModel: PlantDetailViewModel = viewModel(
+        factory = InjectorUtils.providePlantDetailViewModelFactory(ContextAmbient.current, plantId)
+    )
+    val plant = plantDetailsViewModel.plant.observeAsState().value
+    val isPlanted = plantDetailsViewModel.isPlanted.observeAsState().value
+
+    // Every time there's a new value for plant or isPlanted LiveData, this block will get executed
+    if (plant != null && isPlanted != null) {
+        Surface {
+            var showSnackbar by state { false }
+
+            TextSnackbarContainer(
+                snackbarText = stringResource(R.string.added_plant_to_garden),
+                showSnackbar = showSnackbar,
+                onDismissSnackbar = { showSnackbar = false }
+            ) {
+                val context = ContextAmbient.current
+                PlantDetails(plant, isPlanted, PlantDetailsCallbacks(
+                    onBackClicked = onBackClicked,
+                    onFabClicked = {
+                        plantDetailsViewModel.addPlantToGarden()
+                        showSnackbar = true
```

Comment:
If the state moves into the view model, this logic can move with it. Bumping to strong suggestion.